### PR TITLE
Visual Cues + Money Fix

### PR DESCRIPTION
**Pull request by ToniRib**
Fixed the error where money was being collected for enemies that go off the right of the screen but aren't killed. I only had to add one if statement to the rewardMonies method in game.js to check if the enemy was active prior to adding to the game monies.

Also does some general refactoring in engine.js. I broke out an image-loader.js file to load all the images so that cut down on the number of lines.

Build Sites highlight when they are clicked, but unhighlight on the next click or if a tower is put on them. If there is already a tower on them, they will not highlight.

### CODE REVIEW

**Comment by ToniRib** [comment]
Added a feature where a div is shown that prompts you to click a tower if a build site is highlighted.

@stevekinney or @rrgayhart do you guys have any idea how to refactor lines 113 - 164 of `engine.js`?

**Comment by ToniRib** [comment]
Ok this branch is going to need some refactoring but we can merge it in for now.

**Comment by Jlawlzz** [comment]
Looks awesome! Cant believe how much you got done, the highlighting of tiles is great!

File: lib/engine.js [hunk 1]
```diff
@@ -150,48 +136,84 @@ function gameWin() {
...
 
 function selectTile(event) {
   let clickPosition = getPosition(event);
+  lastTile = currentTile;
   currentTile = board.getClickedTile(clickPosition.x, clickPosition.y);
+
+  if (tileIsABuildTile(currentTile) && tileIsAHighlightedBuildTile(lastTile)) {
+    lastTile.type = 'buildTile';
+    currentTile.type = 'buildTileHighlighted';
+    $('#tower-prompt').show();
```
Comment (rrgayhart) [inline lib/engine.js]: There's a lot of repetition here - so I'd recommend breaking out the prompt show and hide into its own function

**Pull request merged**